### PR TITLE
Chore : Fixing CI

### PR DESCRIPTION
**Pull request by brustolin**
This PR changes two things:

- Remove UITests for iOS 12 from running in the GH simulators. We still run it with SauceLabs.
- Remove Unit tests for view hierarchy from running on iOS 12 tests. GH is hanging when running this tests. We have view hierarchy enabled for UI tests that run in Saucelabs and iOS 12.

_#skip-changelog_

### CODE REVIEW

**Review by philipphofmann — APPROVED** [review]
LGTM

File: Tests/SentryTests/SentryViewHierarchyTests.swift
Comment (brustolin) [reply Tests/SentryTests/SentryViewHierarchyTests.swift:37]: Im still testing, but it looks like the 13 also have the problem

Comment (armcknight) [reply Tests/SentryTests/SentryViewHierarchyTests.swift:37]: You might be able to do this in an `setUpWithError` override instead of inserting it at the beginning of each test method, which diffs every line due to indentation and could be forgotten for new test methods added in the future.

```swift
override func setUpWithError() throws {
	try super.setUpWithError()
	guard #available(iOS 14, *) else {
		throw XCTSkip("Skipping for iOS >14")
	}
}
```

**Review by armcknight — COMMENTED** [review]
Great plan, no need to run the extra compute! A couple suggestions for you.

File: Tests/SentryTests/Integrations/Performance/AppStartTracking/SentryAppStartTrackingIntegrationTests.swift
Comment (brustolin) [inline Tests/SentryTests/Integrations/Performance/AppStartTracking/SentryAppStartTrackingIntegrationTests.swift:1]: The only thing `SentryAppStartTrackingIntegration` do is starting an `SentryAppStartTracker`.
I changed this tests to verify it the integration is properly setting up the tracker. 
Tests for `SentryAppStartTracker` behavior are being done in `SentryAppStartTrackerTests`, here we have no control over `SentrySysctl` which causes flake tests.

File: Sentry.xcodeproj/xcshareddata/xcschemes/Sentry.xcscheme
Comment (philipphofmann) [inline Sentry.xcodeproj/xcshareddata/xcschemes/Sentry.xcscheme:92]: `m`: Why do we need to skip this?

File: Tests/SentryTests/SentryViewHierarchyTests.swift
Comment (philipphofmann) [inline Tests/SentryTests/SentryViewHierarchyTests.swift:34]: TIL 😃

File: Tests/SentryTests/Integrations/Performance/AppStartTracking/SentryAppStartTrackingIntegrationTests.swift
Comment (philipphofmann) [reply Tests/SentryTests/Integrations/Performance/AppStartTracking/SentryAppStartTrackingIntegrationTests.swift:1]: @brustolin, this PR already improves CI a lot. Why not merge it and fix the issues in `SentryAppStartTrackerTests` in an extra PR?

Comment (brustolin) [reply Tests/SentryTests/Integrations/Performance/AppStartTracking/SentryAppStartTrackingIntegrationTests.swift:1]: I was frustrating because CI was not green. I guess we could have separated in different PRs, but since the change is already here I don't think it worth going back.

File: Sentry.xcodeproj/xcshareddata/xcschemes/Sentry.xcscheme
Comment (brustolin) [reply Sentry.xcodeproj/xcshareddata/xcschemes/Sentry.xcscheme]: Im still doing tests. Im not 100% confident that the changes applied here are reliable.

I'll let you know when I think we can merge this for a last review.

File: Tests/SentryTests/Integrations/Performance/AppStartTracking/SentryAppStartTrackingIntegrationTests.swift
Comment (philipphofmann) [inline Tests/SentryTests/Integrations/Performance/AppStartTracking/SentryAppStartTrackingIntegrationTests.swift:50]: `l`:
```suggestion
    func testUnistall_stopsTracker() throws {
```

File: Sources/Sentry/SentryAppStartTracker.m
Comment (philipphofmann) [inline Sources/Sentry/SentryAppStartTracker.m:40]: `m`: This class is internal; I would make this a public readonly property for SentryAppStartTracker. Then we don't need to use Dynamic for testing. You can keep the `Test` compiler directive, but I think removing it would also be fine.

**Review by philipphofmann — APPROVED** [review]
One minor comment. Thanks @brustolin 👏

**Pull request merged**
